src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,166 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> t = tail;
+        Node<T> newNode = new Node<>(t, value, null);
+        if (t == null) {
+            head = newNode;
+        } else {
+            t.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAddedElement(index);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> succ = head;
+            int i = 0;
+            while (index != i) {
+                succ = succ.next;
+                i++;
+            }
+            Node<T> pred = succ.prev;
+            Node<T> newNode = new Node<>(pred, value, succ);
+            succ.prev = newNode;
+            if (pred == null) {
+                head = newNode;
+            } else {
+                pred.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currNode;
+        if (index < (size >> 1)) {
+            currNode = head;
+            for (int i = 0; i < index; i++) {
+                currNode = currNode.next;
+            }
+        } else {
+            currNode = tail;
+            for (int i = size - 1; i > index; i--) {
+                currNode = currNode.prev;
+            }
+        }
+        return currNode.item;
     }
 
     @Override
     public T set(T value, int index) {
+        checkIndex(index);
+        Node<T> currNode = head;
+        for (int i = 0; i <= index; i++) {
+            if (index == i) {
+                T oldItem = currNode.item;
+                currNode.item = value;
+                return oldItem;
+            }
+            currNode = currNode.next;
+        }
         return null;
```

Comment:
The return statement `return null;` is unreachable because the loop will always return a value when the index is found. Consider removing this line.